2.0/m_timedstaticquit.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2017 bigfoot547 <bigfoot@bigfootslair.net>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: bigfoot547 */
+/* $ModAuthorMail: bigfoot@bigfootslair.net */
+/* $ModDesc: Replaces the quit message of a quitting user if they have been connected for less than a configurable time */
+/* $ModDepends: core 2.0 */
+/* $ModConfig: <timedstaticquit quitmsg="Client Quit" mintime="300"> */
+
+#include "inspircd.h"
+
+class ModuleTimedStaticQuit : public Module
+{
+private:
+	std::string quitmsg;
+	int mintime;
+
+public:
+	void init()
+	{
+		OnRehash(NULL);
+		Implementation eventlist[] = { I_OnPreCommand, I_OnRehash };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
+	}
+
+	ModResult OnPreCommand(std::string &command, std::vector<std::string>& parameters, LocalUser *user, bool validated, const std::string &original_line)
```

Comment:
&* should go against the type not the name (and if you're not using the parameter you can omit the name).